lib/facter/util/wmi.rb

```diff
@@ -2,7 +2,10 @@ module Facter::Util::WMI
   class << self
     def connect(uri = wmi_resource_uri)
       require 'win32ole'
-      WIN32OLE.connect(uri)
+      #WIN32OLE.connect(uri)
+      swbem = WIN32OLE.new("WbemScripting.SWbemLocator")
```

Comment:
Odd - there are no other places that Ruby directly calls `CoInitialize` / `CoInitializeEx` / `OleInitialize` / `OleInitializeEx` that I see.  So how is Ruby able to init COM then?


I wrote a trivial script in Ruby demonstrating that you are correct - this basic approach works:


```ruby
require 'win32ole'
swbem = WIN32OLE.new('WbemScripting.SWbemLocator')
server = swbem.ConnectServer('.', 'root\\cimv2')

results = server.ExecQuery('SELECT manufacturer, model FROM Win32_ComputerSystem')
for result in results
  puts result.manufacturer
  puts result.model
end
```

Output:

```
[10.0.16.145]: PS C:\puppet> ruby --version
ruby 2.1.8p440 (2015-12-16 revision 53160) [x64-mingw32]

[10.0.16.145]: PS C:\puppet> ruby .\wmi-com-basic.rb
Microsoft Corporation
Virtual Machine
```